lib/client.js

```diff
@@ -251,6 +251,12 @@ Resolve.prototype._fillServers = function() {
 
   slist = platform.name_servers;
 
+  if (slist.length === 0) {
+    // No nameservers defined on platform, so this._server_list
+    // stays empty.
+    return;
+  }
```

Comment:
The expression `slist[tries % slist.length]` used in the attempt-loop will always result in undefined if `slist.length === 0`. That means `s.address` will fail with a null-reference exception.

After this change `_server_list` stays empty without throwing an exception.

If someone knows a cleaner solution for empty server lists let me know!